robottelo/containers.py

```diff
@@ -0,0 +1,249 @@
+
+    def logs(self, file="stdout", tailing=False):
+        """Enables logging to a file. Defaults to writing to stdout.
+        Parameters:
+            file (str): A file name to write to.
+            tailing (boolian): If set to True, tails the log file.
+        """
+        if file == "stdout":
+            current = self._client.logs(self._inst["Id"])
+        else:
+            current = self.execute(f"cat {file}")
+        if isinstance(current, bytes):
+            current = current.decode()
+        self._logs[file] = current
+        return current.replace(
+            self._logs.get(file, ""), "") if tailing else current
+
+    def port(self):
+        """The port to use for SSH."""
+        return self._client.port(self._inst["Id"], 22)[0]["HostPort"]
+
+
+class Container:
+    """Generic class to gather information for a Linux Container.
+    It enables separating generic and runtime specific code.
+    Calls a runtime specific class to build the container. Defaults to docker.
+    Attributes:
+        runtime (init): The Linux container runtime to use. Defaults to docker.
+        image (init): The name of the container image to build from.
+        tag (init): The tag of the operating system to use.
+        agent (boolian): If set to True, prepares container for katello-agent.
+        port (init): The port to use for SSH.
+    Returns:
+        A container object
+    """
+
+    def __init__(
+            self, runtime="docker", image="ch-d", tag="rhel7", agent=False,
+            ports=None
+    ):
+        if runtime == "docker":
+            ContainerClass = DockerContainer
+        # elif runtime == "podman":
+        #     ContainerClass = PodmanContainer
+        else:
+            logger.warning("Invalid runtime selection")
+        self._inst = ContainerClass(image=image, tag=tag, agent=agent,
+                                    ports=ports)
+        self._logs = self._inst._logs
+
+    @property
+    def name(self):
+        return self._inst.name
+
+    def delete(self):
+        return self._inst.delete()
+
+    def execute(self, command):
+        return self._inst.execute(command)
+
+    def logs(self, file="stdout", tailing=False):
+        return self._inst.logs(file, tailing)
+
+    def port(self):
+        return self._inst.port()
+
+    def register(
+            self,
+            host=settings.server.hostname,
+            ak=None,
+            org="Default_Organization",
+            env="Library",
+            auth=("admin", "changeme"),
+            force=False,
+    ):
+        self._host = host
+        res = self._inst.execute(
+            f"curl --insecure --output katello-ca-consumer-latest.noarch.rpm \
+                https://{host}/pub/katello-ca-consumer-latest.noarch.rpm"
+        )
+        res += self._inst.execute(
+            "yum -y localinstall katello-ca-consumer-latest.noarch.rpm"
+        )
+        if "Complete!" not in res:
```

Comment:
Can we check status code of command running in container, instead of relaying on specific string?